libs/langgraph/langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -629,11 +673,44 @@ async def acall_model(state: AgentState, config: RunnableConfig) -> AgentState:
         # We return a list, because this will get added to the existing list
         return {"messages": [response]}
 
+    def generate_structured_response(
+        state: AgentState, config: RunnableConfig
+    ) -> AgentState:
+        model_with_structured_output = _get_model(model).with_structured_output(
+            cast(Union[dict, type[BaseModel]], response_format)
+        )
+        # NOTE: we exclude the last message because there is enough information
+        # for the LLM to generate the structured response
+        response = model_with_structured_output.invoke(state["messages"][:-1], config)
```

Comment:
Is there a downside in sending the full history? I am worried that this could cause bugs in the future.